ci/openshift.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+
+# This workflow will use Source-to-Image (S2I) to deploy your application to your OpenShift Cluster.
+#
+# To configure this workflow:
+#
+#  1. Ensure that your repository contains a valid Dockerfile
+#
+#  2. Set up secrets in your workspace:
+#     - OPENSHIFT_SERVER_URL with the URL of the Openshift cluster
+#     - API_TOKEN with the API token used for authentication
+#  3. Change the value for Project below to match the project you want to deploy to
+#
+#
+
+name: OpenShift
+
+on:
+  push:
+    branches:
+      - master
+
+# Environment variables available to all jobs and steps in this workflow
+env:
+  Project: dev
+
+jobs:
+  setup-build-publish-deploy:
+    name: Setup, Build, Publish, and Deploy
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      
+    # Create the deployment using Source-to-Image with docker build strategy
+    - name: Deploy using Source-to-Image
+      uses: redhat-developer/openshift-actions@v1.1
```

Comment:
We will have to use a 40 char SHA here and not the version as noted in the guidelines.